capture/crawler_estadao.py

```diff
@@ -0,0 +1,247 @@
+import sys
+import pymongo
+import logging
+import datetime
+import requests
+import settings
+
+from soupy import Soupy, Q
+from bs4 import BeautifulSoup
+from logging.handlers import RotatingFileHandler
+from downloader import compress_content, detect_language
+
+###########################
+#  Setting up Logging
+###########################
+
+logger = logging.getLogger("Estadao")
+logger.setLevel(logging.DEBUG)
+
+# create console handler and set level to debug
+ch = logging.StreamHandler()
+ch.setLevel(logging.DEBUG)
+fh = RotatingFileHandler('/tmp/mediacloud_estadao.log', maxBytes=5e6, backupCount=3)
+
+# create formatter
+formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+
+# add formatter to ch
+ch.setFormatter(formatter)
+fh.setFormatter(formatter)
+
+# add ch to logger
+logger.addHandler(ch)  # uncomment for console output of messages
+logger.addHandler(fh)
+
+client = pymongo.MongoClient(settings.MONGOHOST, 27017)
+MCDB = client.MCDB
+ARTICLES = MCDB.articles  # Article Collection
+ARTICLES.ensure_index("source")
+
+
+def find_articles(category, page=1):
+	"""Get the urls of last news and its categories
+
+	:param category: the category of the news (politica, economia, internacional, esportes, sao-paulo, cultura)
+	:param page: the page number of last news
+	:type category: string
+	:type page: integer
+	:return: last news with its categories
+	:rtype: list()
+
+	"""
+
+	CATEGORIES = (u'politica', u'economia', u'internacional', u'esportes', u'sao-paulo', u'cultura', u'opiniao',
+				  u'alias', u'brasil', u'ciencia', u'educacao', u'saude', u'sustentabilidade', u'viagem')
+
+	if category not in CATEGORIES:
+		raise ValueError("Category value not accepted.")
+
+	params = (category, page)
+
+	INDEX_URL = "http://{0}.estadao.com.br/ultimas/{1}".format(*params)
+
+	index = requests.get(INDEX_URL).content
+	soup = BeautifulSoup(index)
+	news_index = soup.findAll("div", {"class":"listadesc"})
+	news_urls = [url.contents[1]['href'] for url in news_index]
+	
+	return news_urls
+
+def clean_content(response_content):
```

Comment:
Sometimes the Goose module do not work so well. I don't know if this is
the case of Estadão.

On 20-07-2015 18:59, Lucas Machado wrote:
>
> In capture/crawler_estadao.py
> <https://github.com/NAMD/mediacloud_backend/pull/82#discussion_r35049477>:
>
> > +
> > +	if category not in CATEGORIES:
> > +		raise ValueError("Category value not accepted.")
> > +
> > +	params = (category, page)
> > +
> > +	INDEX_URL = "http://{0}.estadao.com.br/ultimas/{1}".format(*params)
> > +
> > +	index = requests.get(INDEX_URL).content
> > +	soup = BeautifulSoup(index)
> > +	news_index = soup.findAll("div", {"class":"listadesc"})
> > +	news_urls = [url.contents[1]['href'] for url in news_index]
> > +	
> > +	return news_urls
> > +
> > +def clean_content(response_content):
>
> All right. I'll try out the goose module.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NAMD/mediacloud_backend/pull/82/files#r35049477>.
>

-- 
Renato